### ISSUE TITLE
Perform the v2.0.0 release ceremony

### ISSUE THREAD

**Issue by defuse** [defuse/php-encryption#47]
I sign the git tags, but we need GPG signatures that people actually know how to verify. For now, I'll use my email address key, and then we'll switch to my air-gapped key in the future once I get it set up correctly.

**Comment by defuse** [comment]
Cross-reference with #44 and #45 to secure the pipeline with signatures from upstream.

**Comment by defuse** [comment]
Well, it doesn't look like composer/packagist support any sort of signature checking at all... *sigh*.

**Comment by mathewpeterson** [comment]
Maybe you could do something similar here: http://fabien.potencier.org/article/73/signing-project-releases

**Comment by defuse** [comment]
I set up a Qubes laptop for my job and I made a VM for codesigning. Just need to generate some keys and then I can make progress on this. I'd also like @paragonie-scott to sign releases (either 1/2 or 2/2 sigs required, not sure yet), so we should talk about the right way to do that.

**Comment by defuse** [comment]
As part of this ticket, I really want to try and make sure installing with composer actually checks the signatures... hopefully composer supports code signing.

**Comment by defuse** [comment]
Is there a "best practices of code signing" document on the web somewhere? Surely there's something that describes the procedure that Debian/Arch/Bitcoin core devs follow? I should do some research instead of rolling my own.

**Comment by paragonie-scott** [comment]
I think one of the problems plaguing the Internet is that such a best practice document does not, in fact, exist.

**Comment by pnowosie** [comment]
@keybase has code-signing feature in it's node client, but now I can see [it's deprecated](https://keybase.io/docs/command_line/code_signing). I mentioned this because they promises something better is coming.

**Comment by defuse** [comment]
Useful comments on this PR: https://github.com/defuse/php-encryption/pull/139

**Comment by defuse** [comment]
To close this ticket:

1. [ ] Add this list to the release process documentation. Make sure public keys are easily found.
2. [ ] Do a dry run of this release process.
3. [ ] Build, commit, and sign a .phar, and commit the signatures.
4. [ ] Sign the git tag for v2.0.0.

This will be the last ticket before v2.0.0 is released, so I'm renaming it accordingly.

**Comment by defuse** [comment]
For simplicity / quickness I'm probably just going to sign everything with my email key, rather than having multiple signatures (since that's confusing to verify). It's also printed on my business cards which I can hand out to people, and relatively unlikely to be compromised.

**Comment by defuse** [comment]
(Oh, and I implicitly decided by writing that numbered list above that I won't do anything special to help composer check signatures... If composer implements such a feature, e.g. by requesting signatures be available in certain paths/filenames then I'll happily comply).

**Comment by defuse** [comment]
I created a milestone for the internal audit. This ticket blocks on that milestone being complete.

**Comment by defuse** [comment]
I automated this so once the audit work is done I just have to run 2 commands to do it :)

**Comment by defuse** [comment]
All ready to go, just blocking on #257 (PR #258), maybe I'll have time for that today!

**Comment by defuse** [comment]
Okay, #264 needs review, and once that's merged I'll publish a v2.0.0-beta (as a practice), and then do the actual v2.0.0 release.

**Comment by glensc** [comment]
someone do `s/cerimony/ceremony/` in ticket title

**Comment by defuse** [comment]
TODO: verify by diffing between `v2.0` and `master` that the merge I did left everything as it was in `v2.0`.

**Comment by paragonie-scott** [comment]
https://github.com/defuse/php-encryption/compare/v2.0...master Nothing changed.

**Comment by defuse** [comment]
TODO: delete the v2.0 branch and all the other dev branches

**Comment by paragonie-scott** [comment]
https://github.com/defuse/php-encryption/compare/e42d3a9ffe924ba005200bbdb4bed675c278954c...606645d1826312f111c9f761c7d994d890766042 is a more robust record of "nothing changed" for anyone in the future who wants to verify that we did v2.0.0's release safely and correctly.

**Comment by paragonie-scott** [comment]
https://github.com/defuse/php-encryption/branches I cleaned a bunch of them, left ones I was uncertain of.

**Comment by defuse** [comment]
Oh, I deleted all the branches at the same time :-)

**Comment by defuse** [comment]
We can recreate a `1.2.1` branch if we ever need to backport fixes to 1.2.1.

**Comment by paragonie-scott** [comment]
We have a release tag to reference.

**Comment by defuse** [comment]
Here's the Travis-CI of the revision I'll (probably) be releasing: https://travis-ci.org/defuse/php-encryption/builds/130682653

**Comment by defuse** [comment]
DONE https://github.com/defuse/php-encryption/releases

**Comment by larowlan** [comment]
🎉 congratulations

**Comment by jedisct1** [comment]
Congrats!

**Comment by glensc** [comment]
> Release tag says there's a lot of compatibility and that you should read the README for more details. README mentions nothing of it.
